src/content/docs/guides/development-workflow.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: Development Workflow
+description: Learn how to integrate Shorebird into your development workflow with automated releases and patches.
+---
+
+import RemoteCode from '~/components/RemoteCode.astro';
+import CICheckSnippet from '~/components/guides/development-workflow/CICheckSnippet.astro';
+
+This guide will walk through an opinionated development workflow with Shorebird.
+We'll cover how to automate releasing and patching with Shorebird to streamline
+the development process.
+
+## Prerequisites
+
+This guide assumes you are already familiar with Shorebird and have integrated
+it into your application. Refer to the [getting started](/) instructions for
+more information.
+
+## Overview
+
+Shorebird provides a set of tools that you can use to push over the air updates
+to your customers but it's up to you to decide how to integrate it into your
+development workflow.
+
+While there are many ways to integrate Shorebird, the following workflow is a
+common approach that we recommend:
+
+![development-workflow](https://github.com/user-attachments/assets/59eaefc4-1213-4549-b0d6-38738b065147)
+
+The illustration above, uses [trunk based
+development](https://trunkbaseddevelopment.com) and tags to trigger automated
+release and patch workflows.
+
+The workflow consists of three main phases:
+
+1. **Development**: Developers work on features and bug fixes.
+1. **Release**: When a feature is ready, a release is created and distributed to
+   customers (typically via stores).
+1. **Patch**: If a critical bug is found, a patch is created and distributed to
+   customers immediately via Shorebird.
+
+Let's walk through each phase in more detail.
+
+## Development
+
+Most of the time is spent in the development phase. During this phase,
+developers are building features and fixing bugs.
+
+The development workflow can be broken down into the following steps:
+
+1. Create branch from trunk (main)
+1. Work on task
+1. Open a pull request to merge changes into trunk
+1. Squash and merge after ci checks & code review
```

Comment:
Should we explain what "squash and merge" is?

Reply:
We also probably want to define this the first time it's used instead of later on